modules/sdk-coin-sui/src/lib/transaction.ts

```diff
@@ -302,31 +266,79 @@ export class Transaction extends BaseTransaction {
     const data = Buffer.from(serializedHex, 'hex');
     const trimmedData = new Uint8Array(data.subarray(TYPE_TAG.length));
     const k = bcs.de('TransactionData', trimmedData);
-    this.updateProperTransactionData(k);
+
+    let type: SuiTransactionType;
+    const txDetails: TxDetails = k.kind.Single;
+    if (txDetails.hasOwnProperty('Pay')) {
```

Comment:
:nit:

one way to avoid chained if-else is to put the accepted types into some sort of set and check if the deserialized type is within that set, otherwise throw an error.